### PR TITLE
HelpCenter: add language to HappyChat authentication

### PR DESCRIPTION
**Pull request by arcangelini**
## Proposed Changes

When starting a new chat the Language option was previously not set. This uses the User data to set the language appropriately.

Markup 2022-09-14 at 00 07 54

## Testing Instructions

1. Pull branch and `cd apps/editing-toolkit/ && yarn dev --sync`
2. Go to the editor and open Help Center
3. Start a new chat and make sure the language matches the account language.
4. Change the language on your account
5. Clear cache and reload the editor and start a new chat.
6. Observe the language matches the new language.

### CODE REVIEW

File: packages/happychat-connection/src/use-happychat-auth.ts [hunk 1]
```diff
@@ -13,14 +12,13 @@ export async function requestHappyChatAuth() {
...
 	const url: string = config( 'happychat_url' );
-	const locale = getLocaleSlug();
 
 	const happychatUser: HappychatUser = {
 		signer_user_id: user.ID,
-		locale,
 		groups: [ 'WP.com' ],
 		skills: {
 			product: [ 'WP.com' ],
+			language: [ user.language ],
```
Comment (alshakero) [inline packages/happychat-connection/src/use-happychat-auth.ts:21]: Optional: we probably want to do this. Some locales are obscure, like "gl". 

```js
import { getLanguage } from '@automattic/i18n-utils';
```
```suggestion
			language: [ getLanguage( user.language ).name ],
```

**Review by alshakero — COMMENTED** [review]
Besides the failing test, this is great. Tested and works perfect.

File: packages/happychat-connection/src/use-happychat-auth.ts (hunk 1 above)
Comment (arcangelini) [reply packages/happychat-connection/src/use-happychat-auth.ts:21]: I think HappyChat uses the language code for chat assignment? I thought his helped route chats to HE's who are assigned with that language skill? @klimeryk 

If that is not the case, then you are absolutely right and we should probably make this into a human readable language.

Comment (alshakero) [reply packages/happychat-connection/src/use-happychat-auth.ts:21]: Let's compare with inline-help and see what it sends and mimic that.

Comment (arcangelini) [reply packages/happychat-connection/src/use-happychat-auth.ts:21]: Based on the test in /Automattic/happychat `src/service/state/middlewares/system/test/chat-assignment.js`

It looks like this is expecting `[ 'en' ]`. I didn't follow the flow to see where it is used specifically though.

**Pull request merged**